test/core/filters/client_authority_filter_test.cc

```diff
@@ -0,0 +1,124 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/ext/filters/http/client_authority_filter.h"
+
+#include <gtest/gtest.h>
+
+#include "src/core/lib/resource_quota/resource_quota.h"
+
+namespace grpc_core {
+namespace {
+
+auto* g_memory_allocator = new MemoryAllocator(
+    ResourceQuota::Default()->memory_quota()->CreateMemoryAllocator("test"));
+
+class TestChannelArgs {
+ public:
+  explicit TestChannelArgs(const char* default_authority) {
+    arg_.key = const_cast<char*>(GRPC_ARG_DEFAULT_AUTHORITY);
+    arg_.type = GRPC_ARG_STRING;
+    arg_.value.string = const_cast<char*>(default_authority);
+    args_.num_args = 1;
+    args_.args = &arg_;
+  }
+
+  const grpc_channel_args* args() const { return &args_; }
+
+ private:
+  grpc_arg arg_;
+  grpc_channel_args args_;
+};
+
+TEST(ClientAuthorityFilterTest, DefaultFails) {
+  EXPECT_FALSE(ClientAuthorityFilter::Create(nullptr).ok());
+}
+
+TEST(ClientAuthorityFilterTest, WithArgSucceeds) {
+  EXPECT_TRUE(ClientAuthorityFilter::Create(
```

Comment:
Suggest having this log the status if the expectation fails, so that test failure messages are easier to understand.

Same throughout.

Reply:
done